Sources/AsyncHTTPClient/ConnectionPool/State Machine/HTTPConnectionPool+HTTP2Connections.swift

```diff
@@ -0,0 +1,446 @@
+            /// the pool is establishing a connection. Valid transitions are to: .idle, .available and .closed
+            case starting
+            /// the connection is waiting to retry to establish a connection. Transition to .closed. From .closed
+            /// a new connection state must be created for a retry.
+            case backingOff
+            /// the connection is active and is able to run requests. Valid transitions to: .draining and .closed
+            case active(Connection, maxStreams: Int, usedStreams: Int, lastIdle: NIODeadline)
+            /// the connection is active and is running requests. No new requests must be scheduled.
+            /// Valid transitions to: .draining and .closed
+            case draining(Connection, maxStreams: Int, usedStreams: Int)
+            /// the connection is closed
+            case closed
+        }
+
+        var isActive: Bool {
+            switch self.state {
+            case .starting, .backingOff, .draining, .closed:
+                return false
+            case .active:
+                return true
+            }
+        }
+
+        var isStarting: Bool {
+            switch self.state {
+            case .starting, .backingOff:
+                return true
+            case .active, .draining, .closed:
+                return false
+            }
+        }
+        
+        /// A request can be scheduled on the connection
+        var isAvailable: Bool {
+            switch self.state {
+            case .active(_, let maxStreams, let usedStreams, _):
+                return usedStreams < maxStreams
+            case .starting, .backingOff, .draining, .closed:
+                return false
+            }
+        }
+
+        /// The connection is active but there are no requests running on the connection
+        var isIdle: Bool {
+            switch self.state {
+            case .active(_, _, let usedStreams, _):
+                return usedStreams == 0
+            case .starting, .backingOff, .draining, .closed:
+                return false
+            }
+        }
+        
+        var isClosed: Bool {
+            switch self.state {
+            case .starting, .backingOff, .draining, .active:
+                return false
+            case .closed:
+                return true
+            }
+        }
+
+        var usedAndMaxStreams: (Int, Int)? {
+            switch self.state {
+            case .starting, .backingOff, .draining, .closed:
+                return nil
+            case .active(_, let maxStreams, let usedStreams, _):
+                return (usedStreams, maxStreams)
+            }
+        }
+        
+        var connection: Connection? {
+            switch state {
+            case .starting, .backingOff, .closed:
+                return nil
+            case .active(let connection, _, _, _),
+                    .draining(let connection, _, _):
+                return connection
+            }
+        }
```

Comment:
I don't think, this is something we want. We only want to get access to the underlying connection, if we want to execute an action that requires this. I don't see any usage of the property, either.